pkg/controller/priorityqueue/priorityqueue_test.go

```diff
@@ -300,6 +300,26 @@ var _ = Describe("Controllerworkqueue", func() {
 		Expect(metrics.depth["test"]).To(Equal(map[int]int{0: 2}))
 	})
 
+	// ref: https://github.com/kubernetes-sigs/controller-runtime/issues/3239
+	It("Get from priority queue might get stuck when the priority queue is shut down", func() {
+		q, _ := newQueue()
+
+		q.Add("baz")
+		// shut down
+		q.ShutDown()
+		q.AddWithOpts(AddOpts{After: time.Second}, "foo")
```

Comment:
Looks like Add still adds options to the shutdown queue. Is this the expected behavior?

Reply:
We can't really keep anyone from calling `Add` as it has no return but we can stop acting on it: https://github.com/kubernetes-sigs/controller-runtime/pull/3250